### PR TITLE
Add new firing damages weapon attributes

### PR DESCRIPTION
**Pull request by petervdmeer**
**Feature:** This PR implements the feature request detailed and discussed in #4532

## Feature Details
Added the following firing damages:
- `firing hull` The amount of hull lost (by the firing ship) when firing this weapon.
- `firing shields` The amount of shield lost (by the firing ship) when firing this weapon
- `firing ion` The amount of ionisation that happens (on the firing ship) when firing this weapon
- `firing slowing` The amount of slowing that happens (on the firing ship) when firing this weapon
- `firing disruption` The amount of shield disruption that happens (on the firing ship) when firing this weapon
- `relative firing energy` The amount of energy lost (by the firing ship) when firing this weapon in relation to total energy capacity.
- `relative firing heat` The amount of heat added (on the firing ship) when firing this weapon in relation to total heat capacity.
- `relative firing fuel` The amount of fuel lost (by the firing ship) when firing this weapon in relation to total fuel.
- `relative firing hull` The amount of hull lost (by the firing ship) when firing this weapon in relation to total hull.
- `relative firing shields` The amount of shield lost (by the firing ship) when firing this weapon in relation to total shields

The `energy damage` type was already implemented by Amazinite and the `firing drain` attribute seems a bit redundant if we already have relative energy damage and ion damage (which can be used for very similar effects), so I didn't include those.

## UI Screenshots
Weapons with the new attributes applied (see testing below for the weapons definitions):
ma_blaster
si_blaster

## Usage Examples
Use the new attributes on (new) weapons and notice the damages happen when such weapon is fired.

## Testing Done
Performed a short test with each of the weapons in the attached datafile. Found 1 issue (check for minimum-hull instead of 0 hull) which is solved now.
[overpowered_weapons.txt](https://github.com/endless-sky/endless-sky/files/5744277/overpowered_weapons.txt)

## Performance Impact
The CanFire() function now also needs to deal with the lookups for those new attributes. The calculations are not very complex, so I expect the additional calculation time to be acceptable.

### CODE REVIEW

File: source/OutfitInfoDisplay.cpp [hunk 1]
```diff
@@ -334,7 +344,17 @@ void OutfitInfoDisplay::UpdateAttributes(const Outfit &outfit)
 		outfit.RelativeEnergyDamage() * 100.,
 		outfit.FiringEnergy(),
 		outfit.FiringHeat(),
-		outfit.FiringFuel()
+		outfit.FiringFuel(),
+		outfit.FiringHull(),
+		outfit.FiringShields(),
+		outfit.FiringIon(),
+		outfit.FiringSlowing(),
+		outfit.FiringDisruption(),
```
Comment (Amazinite) [inline source/OutfitInfoDisplay.cpp:352]: Just glancing right now. Status effects should be multiplied by 100.
```suggestion
		outfit.FiringIon() * 100.,
		outfit.FiringSlowing() * 100.,
		outfit.FiringDisruption() * 100.,
```

File: source/Ship.cpp [hunk 2]
```diff
@@ -42,6 +42,7 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 #include <cmath>
 #include <limits>
 #include <sstream>
+#include <iostream>
```
Comment (tatami4) [inline source/Ship.cpp:45]: This isn't needed.

Comment (petervdmeer) [reply source/Ship.cpp:45]: Agreed, this is a left-over from debugging.. I'll remove it.

File: data/tooltips.txt [hunk 3]
```diff
@@ -596,6 +626,33 @@ tip "firing heat / shot:"
 tip "firing fuel / shot:"
 	`Fuel consumed per shot.`
```
Comment (Amazinite) [inline data/tooltips.txt:628]: Looks like we're missing a `"firing hull / shot"` tooltip.

**Review by Amazinite — CHANGES_REQUESTED** [review]

**Review by Amazinite — APPROVED** [review]

**Pull request merged**